### PR TITLE
Propagate function numbers to task callbacks

### PR DESCRIPTION
**Pull request by tomyhoi**
Function numbers were not available at the task creation callback point.  This PR adds the function ID (fid) number as a parameter to several functions.   It also changes chpl_task_taskCallFTable() from an inline function to a full function, removing any functions called chpl_task_taskCall().

### CODE REVIEW

File: runtime/src/comm/none/comm-none.c [hunk 1]
```diff
@@ -504,7 +504,7 @@ void chpl_comm_execute_on_nb(c_nodeid_t node, c_sublocid_t subloc,
   info->arg_size = arg_size;
   if (arg_size)
     chpl_memcpy(&(info->arg), arg, arg_size);
-  chpl_task_startMovedTask((chpl_fn_p)fork_nb_wrapper, (void*)info,
+  chpl_task_startMovedTask(-1, (chpl_fn_p)fork_nb_wrapper, (void*)info,
```
Comment (mppf) [inline runtime/src/comm/none/comm-none.c]: It seems to me we should have a macro defining no FID or FID unknown

**Comment by tomyhoi** [comment]
Tests on mac and Linux with all combinations of fifo/qthreads X gasnet/none.
Tests on NetBSD with all combinations of fifo X gasnet/none.

File: runtime/src/tasks/qthreads/tasks-qthreads.c [hunk 2]
```diff
@@ -785,7 +786,7 @@ void chpl_task_callMain(void (*chpl_main)(void))
     chpl_qthread_wrapper_args_t wrapper_args =
         {chpl_main, NULL, NULL, false,
          PRV_DATA_IMPL_VAL(CHPL_FILE_IDX_MAIN_TASK , 0,
-                           chpl_qthread_process_tls.chpl_data.id, false,
+                           chpl_qthread_process_tls.chpl_data.id, -1, false,
```
Comment (mppf) [inline runtime/src/tasks/qthreads/tasks-qthreads.c]: Should this be using FID_NONE ?

**Comment by mppf** [comment]
@tomyhoi - this PR is fine with me whenever you are satisfied with it.

**Pull request merged**